### PR TITLE
Update to 42.0.2

### PR DESCRIPTION
**Pull request by Jrice1317**
### cryptography 42.0.2

**Destination channel:** {defaults}

### Links

- [PKG-4018](https://anaconda.atlassian.net/browse/PKG-4018) 
- [Upstream repository](https://github.com/pyca/cryptography/tree/42.0.2)
- [Upstream changelog/diff](https://cryptography.io/en/latest/changelog/)
- Relevant dependency PRs:
  - [cryptography-vectors](https://github.com/AnacondaRecipes/cryptography-vectors-feedstock/pull/19)
  - [setuptools-rust](https://github.com/AnacondaRecipes/setuptools-rust-feedstock/pull/5)

### Explanation of changes:

- Updated `version` and `sha256`
- Changed `license_url` to `license_file`
- Updated `setuptools-rust` dependency
- Updated url to not redirect to `en` for non-English readers
- Added `certifi` to satisfy failing test

[PKG-4018]: https://anaconda.atlassian.net/browse/PKG-4018?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by JeanChristopheMorinPerso — CHANGES_REQUESTED** [review]

**Review by JeanChristopheMorinPerso — APPROVED** [review]
Great job @Jrice1317!

**Comment by Jrice1317** [comment]
> Great job @Jrice1317!

Thank you @JeanChristopheMorinPerso! I really appreciate your reviews!

**Review by cbouss — APPROVED** [review]
LGTM

File: abs.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+channels:
+  - https://staging.continuum.io/prefect/fs/cryptography-vectors-feedstock/pr19/961a3d5
+  - https://staging.continuum.io/prefect/fs/setuptools-rust-feedstock/pr5/855001d
```
Comment (JeanChristopheMorinPerso) [inline abs.yaml:3]: @Jrice1317 You should probably create another PR to remove these from here.

Comment (Jrice1317) [reply abs.yaml:3]: Yeah, when the merge build failed, I remembered I forgot about those! Lol thanks

Comment (Jrice1317) [reply abs.yaml:3]: This is yet another reason `cryptography-vectors` should be merged with `cryptography`

**Pull request merged**